### PR TITLE
docs: integrate onvif schemas and validation matrix to swagger

### PR DESCRIPTION
**Pull request by ajcasagrande**
Lots of additional changes/improvements to the swagger generation:

- Link json schemas converted from wsdl to the corresponding endpoints in the swagger file
- Cleanup and fix response types and include base types from `core-command` service
- For `GET` calls with `jsonObject` field, provide documentation on the format and schema of the field, along with descriptions of each field.
- Parse the tested onvif cameras validation matrix and insert it (footnotes and all) into the description of each linked command
- Add useful warning messages during generation if some commands are missing required or optional data
- Switch to `ruamel.yaml` to allow multi-line strings to support Markdown sytanx for description fields.
- Remove unused schema definitions from onvif wsdl
- Auto generate documentation for base64 env vars

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- **[N/A]** I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [X] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

**Browse the hosted version:**
https://app.swaggerhub.com/apis-docs/ANTHONYJINTEL/edge-x_onvif_device_service/2.2.0

## New Dependency Instructions (If applicable)

> Note: these dependencies are only used for the purpose of generating the OpenAPI 3.0 swagger file.

JavaScript:
- [postman-to-openapi](https://www.npmjs.com/package/postman-to-openapi)

Python:
- [PyYAML](https://github.com/yaml/pyyaml)
- [ruamel.yaml](https://pypi.org/project/ruamel.yaml/)

<hr/>

**A few example screenshots:**

jsonObject schema:

tested cameras matrix:

response event/object schemas:

### CODE REVIEW

File: doc/openapi/README.md [hunk 1]
```diff
@@ -1,8 +1,35 @@
-This code generates OpenAPI spec based on the Postman Collection 
+# OpenAPI / Swagger Spec Files
+This code generates OpenAPI 3.0 spec based on the Postman Collection 
 [device-onvif-camera.postman_collection.json](../postman/device-onvif-camera.postman_collection.json). 
 It also does some automated find and replace to insert some missing example values.
 
+> **TODO:** update usage
```
Comment (brianointel) [inline doc/openapi/README.md]: is this TODO still valid?

File: doc/postman/device-onvif-camera.postman_collection.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
 	"info": {
-		"_postman_id": "117aa9ba-01bf-421f-bf59-fb109888c2d4",
-		"name": "device-onvif-camera Copy 4",
+		"_postman_id": "41b06358-d096-4b9b-8735-a0905762330f",
+		"name": "device-onvif-camera Copy 2",
```
Comment (brianointel) [inline doc/postman/device-onvif-camera.postman_collection.json]: should this be renamed to `device-onvif-camera`?

File: doc/openapi/README.md (hunk 1 above)
Comment (ajcasagrande) [reply doc/openapi/README.md]: Yes it was. Fixed now.

File: doc/postman/device-onvif-camera.postman_collection.json (hunk 2 above)
Comment (ajcasagrande) [reply doc/postman/device-onvif-camera.postman_collection.json]: Yes, good catch. Edit: Fixed

File: doc/openapi/python/postprocess.py [hunk 3]
```diff
@@ -20,49 +34,87 @@
...
-    yml = None
-    sidecar = None
-    profile = None
-    resources = {}
+    matrix: MarkdownMatrix
+    postman_env_file: str
+    yml: any = None
+    sidecar: any = None
+    profile: any = None
+    postman_env: dict[str, str] = field(default_factory=dict)
```
Comment (brianointel) [inline doc/openapi/python/postprocess.py]: to get it to work, I had to add '

    postman_env: 'dict[str, str]' = field(default_factory=dict)
    resources: 'dict[str, any]' = field(default_factory=dict)
    wsdl_files: 'dict[str, any]' = field(default_factory=dict)

File: doc/openapi/python/cleaner.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+@dataclass
+class SchemaCleaner:
+    """
+    Implementation of a schema cleaner to remove unused schema definitions.
+
+    Note: This is a one time use object, do not re-use it!
+    """
+    yml: any
+    schemas: dict[str, Schema] = field(default_factory=lambda: defaultdict(Schema))
```
Comment (brianointel) [inline doc/openapi/python/cleaner.py]: to get it to work, i had to add a quote around the type:

    schemas: 'dict[str, Schema]' = field(default_factory=lambda: defaultdict(Schema))
    unused: 'set[str]' = field(default_factory=set)

**Review by brianointel — DISMISSED** [review]
Looks good!

**Review by ChristianDarr-personal — COMMENTED** [review]
I'm still looking through this, but I think that are too many output statements. Maybe an option to disable it or just removing some of the more extraneous ones would be mote helpful.

**Comment by ajcasagrande** [comment]
> I'm still looking through this, but I think that are too many output statements. Maybe an option to disable it or just removing some of the more extraneous ones would be mote helpful.

@ChristianDarr-personal I agree. The prints were helpful when creating the scripts, but now they are just noise. I have switched to using the python `logging` package and have set a majority of them as debug. By default they are off, but can be enabled by doing `DEBUG_LOGGING=1 make ...`.

**Review by ChristianDarr-personal — APPROVED** [review]

**Pull request merged**